### PR TITLE
[toolbars-v3] fix: isVisible function

### PR DESCRIPTION
**Pull request by noamtwix**

### CODE REVIEW

File: packages/ricos-editor/web/src/tiptap/RicosToolbars.tsx [hunk 1]
```diff
@@ -173,9 +173,8 @@ class RicosToolbars extends React.Component<
         <FloatingToolbar
           editor={tiptapEditor}
           portal={ricosContext.portal}
-          isVisible={
-            !this.isSelectionCollapsed(tiptapEditor.state.selection) &&
-            isTextSelection(tiptapEditor.state.selection)
+          isVisible={selection =>
```
Comment (lxgreen) [inline packages/ricos-editor/web/src/tiptap/RicosToolbars.tsx:176]: please move function definition out of prop. this causes lots of redundant renders

File: packages/ricos-editor/web/src/tiptap/RicosToolbars.tsx [hunk 2]
```diff
@@ -204,7 +203,7 @@ class RicosToolbars extends React.Component<
         <FloatingToolbar
           editor={tiptapEditor}
           portal={ricosContext.portal}
-          isVisible={this.isLinkInCollapsedSelection(tiptapEditor.state.selection, editor)}
+          isVisible={selection => this.isLinkInCollapsedSelection(selection, editor)}
```
Comment (lxgreen) [inline packages/ricos-editor/web/src/tiptap/RicosToolbars.tsx:206]: also here

File: packages/ricos-editor/web/src/toolbars/PluginToolbar.tsx [hunk 3]
```diff
@@ -53,7 +53,7 @@ class PluginsToolbar extends React.Component<
       <FloatingToolbar
         editor={tiptapEditor}
         portal={ricosContext.portal}
-        isVisible={isNodeSelection(tiptapEditor.state.selection)}
+        isVisible={selection => isNodeSelection(selection)}
```
Comment (lxgreen) [inline packages/ricos-editor/web/src/toolbars/PluginToolbar.tsx:56]: and here

**Review by lxgreen — CHANGES_REQUESTED** [review]

File: packages/ricos-editor/web/src/toolbars/PluginToolbar.tsx [hunk 4]
```diff
@@ -53,7 +56,7 @@ class PluginsToolbar extends React.Component<
       <FloatingToolbar
         editor={tiptapEditor}
         portal={ricosContext.portal}
-        isVisible={isNodeSelection(tiptapEditor.state.selection)}
+        isVisible={this.pluginToolbarIsVisibleFunction}
```
Comment (lxgreen) [inline packages/ricos-editor/web/src/toolbars/PluginToolbar.tsx:59]:
```suggestion
        isVisible={isNodeSelection}
```

Comment (lxgreen) [reply packages/ricos-editor/web/src/toolbars/PluginToolbar.tsx:59]: worth to check if it works like that

File: packages/ricos-editor/web/src/tiptap/RicosToolbars.tsx [hunk 5]
```diff
@@ -123,6 +123,12 @@ class RicosToolbars extends React.Component<
     return isCollapsed && isLinkInSelection;
   };
 
+  formattingToolbarIsVisibleFunction = (selection: Selection) =>
+    !this.isSelectionCollapsed(selection) && isTextSelection(selection);
```
Comment (lxgreen) [inline packages/ricos-editor/web/src/tiptap/RicosToolbars.tsx:127]:
```suggestion
  formattingToolbarIsVisibleFunction = and(not(this.isSelectionCollapsed), isTextSelection)
```

File: packages/ricos-editor/web/src/tiptap/RicosToolbars.tsx [hunk 6]
```diff
@@ -123,6 +123,12 @@ class RicosToolbars extends React.Component<
...
+
+  linkToolbarIsVisibleFunction = (selection: Selection) =>
```
Comment (lxgreen) [inline packages/ricos-editor/web/src/tiptap/RicosToolbars.tsx:129]: note for all the visibility functions -- what a naming! how about `isXXXToolbarVisible`?

**Review by lxgreen — CHANGES_REQUESTED** [review]
more declarative, please

File: packages/ricos-editor/web/src/tiptap/RicosToolbars.tsx (hunk 5 above)
Comment (noamtwix) [reply packages/ricos-editor/web/src/tiptap/RicosToolbars.tsx:127]: why is it better to do this?

**Pull request merged**